sets/set.go

```diff
@@ -161,9 +161,19 @@ func Intersect[T comparable](sets ...Set[T]) Set[T] {
 // Applies the given function to each element of a set. Returns the resulting
 // set of function outputs.
 func Map[T, U comparable](ts Set[T], f func(T) U) Set[U] {
+	return FilterMap(ts, func(t T) optionals.Optional[U] {
+		return optionals.Some(f(t))
+	})
+}
+
+// Applies the given function to each element of a set. Removes any None results
+// and returns the rest.
+func FilterMap[T, U comparable](ts Set[T], f func(T) optionals.Optional[U]) Set[U] {
```

Comment:
I have a small anti-preference for this name.  It seems analagous to Haskell's mapMaybe, perhaps `MapOptional`? 

"filter" suggests to me that there's a predicate involved.

Reply:
This name is taken from OCaml: https://ocaml.janestreet.com/ocaml-core/v0.12/doc/base/Base/List/index.html#val-filter_map. Indeed, `f` encodes both a predicate and a mapping function simultaneously: `map h (filter g l)` is equivalent to
```
filter_map (fn x => if g x then Some (h x) else None) l
```